wait.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+)
+
+// Wait for hosts to become ready
+func Wait(hosts []string, timeout int) {
+	begin := time.Now()
+	var wg sync.WaitGroup
+	cancel := make(chan struct{})
+
+	for _, host := range hosts {
+		wg.Add(1)
+		go func(host string) {
+			WaitForHTTPRequest(host, cancel)
+			wg.Done()
+		}(fmt.Sprintf("%s:8080", host))
+	}
+
+	timer := time.AfterFunc(time.Duration(timeout)*time.Second, func() {
+		close(cancel)
+	})
+
+	wg.Wait()
+
+	if !timer.Stop() {
+		log.Printf("Error: One or more services timed out after %d second(s)", timeout)
+		os.Exit(1)
+	}
+	fmt.Printf("\nAll services are up after %v!\n", time.Now().Sub(begin))
```

Comment:
any reason why this is `fmt.Printf` instead of `log.Printf` (`fmt.Printf` goes to stdout while `log.Printf` goes to stderr and has a time prefix)